lib/database/sthDatabase.js

```diff
@@ -313,6 +313,60 @@ function getCollection(params, options, callback) {
         return process.nextTick(callback.bind(null, error));
     }
 
+    //Call fetchCollection after creating database connection if connection was not restored after database restarted
+    if (isConnectionAlive()) {
+        fetchCollection(databaseName, isAggregated, shouldTruncate, shouldCreate, collectionName, callback);
+    } else {
+        reconnectDB(callback, function() {
+            fetchCollection(databaseName, isAggregated, shouldTruncate, shouldCreate, collectionName, callback);
+        });
+    }
+}
```

Comment:
We think that reconnection process should be managed by the underlying MongoDB driver (take into account PR https://github.com/telefonicaid/fiware-sth-comet/pull/565) so the `reconnectDB()` logic should be removed. Note that #570 (the issue being solved by this PR) is about returning 500 when the database is not connected, not about dealing with reconnections.

Thus, This block of code should be changed to something like this:

```
    if (isConnectionAlive()) {
        fetchCollection(databaseName, isAggregated, shouldTruncate, shouldCreate, collectionName, callback);
    } else {
        <invoke the callback with the proper error>
    }
}
```

and the `reconnectDB() function should be removed from code.

Reply:
Fixed in commit https://github.com/telefonicaid/fiware-sth-comet/pull/574/commits/d056f0a6875fb6229822e348b361f829d95a730a